New

ChhaSahil opened the pull request:
## Polyglot JS Task Submission
<!-- - [x] mark like this where required -->

- [ ] **Checkpoint 1**: Compare the cards of hacker and player.
> 20pts
- [ ] **Checkpoint 2**: Keep a track of the individual scores of the hacker and player.
> 20pts
- [ ] **Checkpoint 3**: Deploy the code to GitHub Pages.
> 10pts
- [ ] **Extras:** You can improve UI of the game ;).

### Submission

<!-- Add in your repo and apk link or web site link as per track -->
[Repo] : https://github.com/ChhaSahil/Polyglot_JS.git
[Deployed Site] :  https://chhasahil.github.io/Polyglot_JS/

ChhaSahil committed:
Game Redefined

ChhaSahil committed:
Game Redifined

Divyansh013 commented:
@ankur12-1610 can you please check why github action is failing?

ankur12-1610 commented:
> @ankur12-1610 can you please check why github action is failing?

:cry: ig the Github Actions are not able to read user credentials of the forked repos due to some security problem. I'll remove it.

Divyansh013 commented:
@ChhaSahil overall the code looks good to me. Liked your idea of asking player's name. Since you are doing this I will suggest two things. 
1.  If name is empty, it should alert something. 
2.  If player wins, it should write his name instead of "you won" e.g. Sahil won. Something like that. Else there is no sense of asking player's name.
Rest apart, great work. We will check the plagarism part later.
@ankur12-1610 any other suggestions?

ankur12-1610 commented:
> @ChhaSahil overall the code looks good to me. Liked your idea of asking player's name. Since you are doing this I will suggest two things. 
> 1.  If name is empty, it should alert something. 
> 2.  If player wins, it should write his name instead of "you won" e.g. Sahil won. Something like that. Else there is no sense of asking player's name.
> Rest apart, great work. We will check the plagarism part later.
> @ankur12-1610 any other suggestions?

Yeah, I agree upon that, the implementation of inserting name was good but as @Divyansh013 said we if name field remains empty throwing an error will make this even more better.

Secondly, displaying the name of the player will make it fell more personal and connected. Something `Sahil great work, you successfully tackled all the hacking attempts by the hacker`.

Rest all is good :) Great work.

ChhaSahil commented:
> @ChhaSahil overall the code looks good to me. Liked your idea of asking player's name. Since you are doing this I will suggest two things.
> 
> 1. If name is empty, it should alert something.
> 2. If player wins, it should write his name instead of "you won" e.g. Sahil won. Something like that. Else there is no sense of asking player's name.
>    Rest apart, great work. We will check the plagarism part later.
>    @ankur12-1610 any other suggestions?

Thank you bhaiya ....I will be implementing your suggestions as early as possible 🙏

ChhaSahil committed:
Minor Changes

ChhaSahil commented:
@Divyansh013  @ankur12-1610  bhaiya i guess I have implemented what to asked me to do :) . Sorry I was late due to other commitments . Thank you again!!

ankur12-1610 commented:
> @Divyansh013  @ankur12-1610  bhaiya i guess I have implemented what to asked me to do :) . Sorry I was late due to other commitments . Thank you again!!

Great work, there's one thing I don't know if it is trivial or not, but the name popup shouldn't contain `cancel` button, I'll tell you why: we are throwing an error If the name field is empty which is perfect but when we do cancel the name is not entered default value is taken which makes the popup non trivial. 

If the popup only has enter button it'll be much more precise;)

Rest LGTM!

Great work 🙌

ChhaSahil commented:
> > @Divyansh013  @ankur12-1610  bhaiya i guess I have implemented what to asked me to do :) . Sorry I was late due to other commitments . Thank you again!!
> 
> Great work, there's one thing I don't know if it is trivial or not, but the name popup shouldn't contain `cancel` button, I'll tell you why: we are throwing an error If the name field is empty which is perfect but when we do cancel the name is not entered default value is taken which makes the popup non trivial.
> 
> If the popup only has enter button it'll be much more precise;)
> 
> Rest LGTM!
> 
> Great work 🙌

Yes @ankur12-1610  bhaiya ....working on your suggestion ,expected to implement that too by evening . Thanks again

ChhaSahil committed:
another commit

ChhaSahil commented:
> > > @Divyansh013  @ankur12-1610  bhaiya i guess I have implemented what to asked me to do :) . Sorry I was late due to other commitments . Thank you again!!
> > 
> > 
> > Great work, there's one thing I don't know if it is trivial or not, but the name popup shouldn't contain `cancel` button, I'll tell you why: we are throwing an error If the name field is empty which is perfect but when we do cancel the name is not entered default value is taken which makes the popup non trivial.
> > If the popup only has enter button it'll be much more precise;)
> > Rest LGTM!
> > Great work 🙌
> 
> Yes @ankur12-1610 bhaiya ....working on your suggestion ,expected to implement that too by evening . Thanks again.

@ankur12-1610  bhaiya  I have fixed the issue raised by you in a manner that I have kept the Cancel button ,but if the user clicks on cancel button , It will again prompt the user to enter valid name :).please check,
